components/FilesViewerModal.tsx

```diff
@@ -0,0 +1,339 @@
+  }
+`;
+
+const Header = styled.div`
+  position: absolute;
+  top: 0;
+  right: 0;
+  left: 0;
+  z-index: 3500;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  color: white;
+  background: linear-gradient(to bottom, rgba(0, 0, 0, 0.65) 25%, transparent 100%);
+  height: 74px;
+  padding: 16px;
+`;
+
+const ContentWrapper = styled.div`
+  max-width: 100%;
+  width: 100%;
+  justify-content: center;
+  display: flex;
+`;
+
+const Content = styled.div`
+  max-width: 100%;
+  z-index: 3000;
+`;
+
+const Scroller = styled.div`
+  padding: 64px 16px;
+  overflow-y: auto;
+  max-height: 100vh;
+  width: 100%;
+  display: flex;
+  justify-content: center;
+`;
+
+export const ModalOverlay = styled.button`
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background: rgba(0, 0, 0, 0.7);
+  z-index: 2500;
+  display: block;
+  animation: ${fadeIn} 0.25s;
+  will-change: opacity;
+`;
+
+type WrapperProps = {
+  zindex?: number;
+};
+
+const Wrapper = styled(Flex)<WrapperProps>`
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100vw;
+  height: 100vh;
+  z-index: ${props => props.zindex || 3000};
+
+  justify-content: center;
+  align-items: center;
+`;
+
+const StyledArrowButton = styled.button<{
+  disabled?: boolean;
+  direction?: 'left' | 'right';
+}>`
+  outline: none;
+  color: white;
+  background: rgba(0, 0, 0, 0.7);
+  border: none;
+  position: absolute;
+  z-index: 3001;
+  border-radius: 999px;
+  height: 40px;
+  width: 40px;
+  margin: 16px;
+  cursor: pointer;
+
+  display: ${props => (props.disabled ? 'none' : 'block')};
+  ${props =>
+    props.direction &&
+    css`
+      position: absolute;
+
+      ${props.direction === 'left' ? 'left: 0' : 'right: 0'};
+    `}
+  &:hover {
+    background: ${themeGet('colors.primary.700')};
+  }
+  &:focus {
+    background: ${themeGet('colors.primary.700')};
+  }
+`;
+
+const Button = styled.button`
+  color: white;
+  border: none;
+  outline: none;
+  background: transparent;
+  z-index: 3000;
+  padding: 8px;
+  border-radius: 99px;
+  &:hover {
+    background: black;
+    color: white;
+  }
+  &:focus-visible {
+    background: black;
+    color: white;
+  }
+  cursor: pointer;
+`;
+
+const ButtonLink = styled.a`
+  display: block;
+  color: white;
+  border: none;
+  outline: none;
+  background: transparent;
+  z-index: 3000;
+  padding: 8px;
+  border-radius: 99px;
+
+  &:hover {
+    background: black;
+    color: white;
+  }
+
+  &:focus-visible {
+    background: black;
+    color: white;
+  }
+  cursor: pointer;
+`;
+
+const StyledImg = styled.img`
+  width: 100%;
+`;
+
+FilesViewerModal.propTypes = {
+  onClose: PropTypes.func.isRequired,
+  parentTitle: PropTypes.string,
+  /** Specifies what file url to open first (opens first if undefined) */
+  openFileUrl: PropTypes.string,
+  files: PropTypes.arrayOf(
+    PropTypes.shape({
+      url: PropTypes.string,
+      name: PropTypes.string,
+    }),
+  ),
+};
```

Comment:
:ice_cream: (No change requested) I prefer to use TS only (no proptypes) in TS files. Any reason to keep them here? On a sidenote, it's strange to assign proptypes before declaration.

Reply:
Ah, good call, switched to type definitions instead. Left over from the previous FilesPreviewModal that served as a starting point.